tests/gatewayapitests/utils.go

```diff
@@ -126,19 +126,21 @@ func UnsetListenerHostname(l *gatewayv1.Listener) {
 
 func GetListenersV1(ports []int32, samehost bool, secrets ...string) []gatewayv1.Listener {
 	listeners := make([]gatewayv1.Listener, 0, len(ports))
-	hostname := ""
+
 	for _, port := range ports {
-		if !samehost {
-			hostname = fmt.Sprintf("foo-%d.com", port)
-		} else {
-			hostname = "foo.com"
-		}
 		listener := gatewayv1.Listener{
 			Name:     gatewayv1.SectionName(fmt.Sprintf("listener-%d", port)),
 			Port:     gatewayv1.PortNumber(port),
 			Protocol: gatewayv1.ProtocolType("HTTPS"),
-			Hostname: (*gatewayv1.Hostname)(&hostname),
 		}
+		if !samehost {
+			hostname := fmt.Sprintf("foo-%d.com", port)
+			listener.Hostname = (*gatewayv1.Hostname)(&hostname)
```

Comment:
this line can be taken out from if:else ladder

Reply:
Could you please explain how is this change fixing the ut

Reply:
@pkoshtavmware we are using address of the variable. so previously, all hostname entries in multiple listener scenario was getting overwritten with last hostname in the list. 
So now for each entry, we are allocating new variable and assigning it. Scope of that variable is limited to if loop. So added that logic inside if and else.